build-docs.sh

```diff
@@ -0,0 +1,26 @@
+#! /bin/bash
+
+REMOTE=origin
+
+set -e
+
+if ! [ -f "docs/src/SUMMARY.md" ]; then
+    echo "Run this from the root of the repo"
+    exit 1
+fi
+
+# build the latest version of the mdbook plugin
+cargo build -p taskchampion-cli --features usage-docs --bin usage-docs
+
+if ! [ -d ./docs/tmp ]; then
+    git worktree add docs/tmp gh-pages
```

Comment:
Might be worth adding a few comments to what each of these are doing - all makes perfect sense, just a bit hard to determine what each line is doing on first glance (e.g "make working copy of gh-pages in doc/tmp/", "remove old generated docs" etc)